engine/flink/tests/src/test/scala/pl/touk/nussknacker/devmodel/SqlKafkaPingPongTest.scala

```diff
@@ -89,4 +154,87 @@ class SqlKafkaPingPongTest extends FlinkWithKafkaSuite {
     }
   }
 
+  test("should ping-pong with sql kafka source and sql kafka sink") {
+    val topics = createAndRegisterTopicConfig(topicNaming2, simpleTypesSchema)
+
+    sendAsJson(record1, topics.input)
+    sendAsJson(record2, topics.input)
+
+    val scenarioId = "scenarioId"
+    val sourceId   = "input"
+    val process = ScenarioBuilder
+      .streaming(scenarioId)
+      .parallelism(1)
+      .source(sourceId, "tableApi-source-sql", SqlComponentFactory.TableNameParamName -> s"'$sqlInputTableNameTest2'")
+      .filter("filterId", "#input.someInt != 1")
+      .emptySink(
+        "end",
+        "tableApi-sink-sql",
+        SqlComponentFactory.TableNameParamName -> s"'$sqlOutputTableNameTest2'",
+        SqlSinkFactory.ValueParamName          -> "#input"
+      )
+
+    run(process) {
+      val result = kafkaClient
+        .createConsumer()
+        .consumeWithJson[Json](topics.output)
+        .take(1)
+        .map(_.message())
+
+      result.head shouldBe parseJson(record2)
+    }
+  }
+
+  test("should pong with explicit spel record and DataStream kafka sink") {
+    val topics = createAndRegisterTopicConfig(topicNaming3, simpleTypesSchema)
+
+    sendAsJson(record1, topics.input)
+
+    val process = ScenarioBuilder
+      .streaming("testScenario")
+      .parallelism(1)
+      .source("start", "tableApi-source-sql", SqlComponentFactory.TableNameParamName -> s"'$sqlInputTableNameTest3'")
+      .emptySink(
+        "end",
+        "tableApi-sink-sql",
+        SqlComponentFactory.TableNameParamName -> s"'$sqlOutputTableNameTest3'",
+        SqlSinkFactory.ValueParamName          -> s"$record2"
```

Comment:
It is tricky that you've used json in spel. I know, json is a valid spel record expression :) Also it is hard to read this test because this main thing in the test is somewhere on the end of test. IMO inlining these records and expressions would be more readable.

Reply:
I've inlined all records - theres duplication now but imo too its more readable